packages/seducer/src/reducer.js

```diff
@@ -0,0 +1,111 @@
+/* eslint-disable no-use-before-define */
+import logger from "./logger";
+
+/**
+ * Summary: Reducer controls all actions and reduce them to create the nextState
+ *
+ * Description. Curry function execute (actions, hasLogger, interceptors) and returns nextState
+ *
+ * (actions, hasLogger, interceptors) =>
+ * @param {{[key]:(state, payload)=> nextState}}   actions         An object with {key: value} format that define an action of an state
+ * @param {boolean}                                hasLogger       Allowed to print a friendly log with the previous and next action
+ * @param {{[key]:(state, payload)=> nextState}}   interceptors    An object with {key: value} which allows inversion of control by extending the actions object
+ *
+ * @return (state, actions) => (state, action) => {...}
+ *
+ * @param {unknown}                                state           The current state of the useReducer state can be anything
+ * @param {{key: {type:string, payload:unknown}}}  action          The action to be handle by the reducer and redirected to the their respective actions
+ *
+ * @return {unknown} nextState
+ */
+
+export const reducer = (actions, hasLogger, interceptors = {}) => (state, action) => {
+  const isAValidAction =
+    "type" in action && ([action.type] in actions || (interceptors && [action.type] in interceptors));
```

Comment:
Is the first checking of `"type" in action` necessary? If someone uglifies the code, `action.type` might be changed to `a.b`
Also maybe I missed something, what's the difference between `[action.type] in actions` and `action.type in actions` ?

Reply:
the regular signature for useReducer will be `dispatch({ type: "add", payload: [] })` as an example.

So I'm checking that the dispatch has the `type` property declared (internal checking). We could assume it will have it since I'm the one dispatching it at https://github.com/acl-services/paprika/blob/UXD-630-migration-from-reducerish/packages/seducer/src/useSeducer.js#L39

So I don't think that one will get minified, then I'm checking that the actions passdown by the user have a property `[action.type]` or if the interceptors has it. 

Do you think I should remove the first checking?